plugins/vpp/aclplugin/vppcalls/interfaces_vppcalls.go

```diff
@@ -33,76 +29,108 @@ type ACLInterfaceLogicalReq struct {
 		aclIndex:  ACLIndex,
 		ifIndices: ifIndices,
 		ingress:   true,
 	})
 }
 
-// SetACLToInterfacesAsEgress sets ACL to all provided interfaces as egress
-func (acl *ACLInterfacesVppCalls) SetACLToInterfacesAsEgress(ACLIndex uint32, ifIndices []uint32) error {
-	acl.log.Debugf("Setting up IP egress ACL from interfaces: %v ", ifIndices)
-
-	return acl.requestSetACLToInterfaces(&ACLInterfaceLogicalReq{
+func (handler *aclVppHandler) SetACLToInterfacesAsEgress(ACLIndex uint32, ifIndices []uint32) error {
+	return handler.requestSetACLToInterfaces(&ACLInterfaceLogicalReq{
 		aclIndex:  ACLIndex,
 		ifIndices: ifIndices,
 		ingress:   false,
 	})
 }
 
-// RemoveIPEgressACLFromInterfaces removes ACL from interfaces
-func (acl *ACLInterfacesVppCalls) RemoveIPEgressACLFromInterfaces(ACLIndex uint32, ifIndices []uint32) error {
-	acl.log.Debugf("Removing IP egress ACL from interfaces: %v ", ifIndices)
-
-	return acl.requestRemoveInterfacesFromACL(&ACLInterfaceLogicalReq{
+func (handler *aclVppHandler) RemoveIPEgressACLFromInterfaces(ACLIndex uint32, ifIndices []uint32) error {
+	return handler.requestRemoveInterfacesFromACL(&ACLInterfaceLogicalReq{
 		aclIndex:  ACLIndex,
 		ifIndices: ifIndices,
 		ingress:   false,
 	})
 }
 
-func (acl *ACLInterfacesVppCalls) requestSetACLToInterfaces(logicalReq *ACLInterfaceLogicalReq) error {
+func (handler *aclVppHandler) SetMacIPAclToInterface(aclIndex uint32, ifIndices []uint32) error {
+	setACLStopwatch := measure.GetTimeLog(acl_api.MacipACLInterfaceAddDel{}, handler.stopwatch)
+	for _, ingressIfIdx := range ifIndices {
+		// Measure MacipACLInterfaceAddDel time
+		start := time.Now()
+
+		req := &acl_api.MacipACLInterfaceAddDel{
+			ACLIndex:  aclIndex,
+			IsAdd:     1,
+			SwIfIndex: ingressIfIdx,
+		}
+
+		reply := &acl_api.MacipACLInterfaceAddDelReply{}
+
+		err := handler.callsChannel.SendRequest(req).ReceiveReply(reply)
+		if err != nil {
+			return fmt.Errorf("failed to set interface %v to L2 ACL %v", ingressIfIdx, aclIndex)
+		}
+		if reply.Retval != 0 {
+			return fmt.Errorf("set interface %v to L2 ACL %v returned %v", ingressIfIdx, aclIndex, reply.Retval)
+		}
+
+		// Log MacipACLInterfaceAddDel time measurement results.
+		if setACLStopwatch != nil {
+			setACLStopwatch.LogTimeEntry(time.Since(start))
+		}
+	}
+
+	return nil
+}
+
+func (handler *aclVppHandler) RemoveMacIPIngressACLFromInterfaces(removedACLIndex uint32, ifIndices []uint32) error {
+	setACLStopwatch := measure.GetTimeLog(acl_api.MacipACLInterfaceAddDel{}, handler.stopwatch)
+	for _, ifIdx := range ifIndices {
+		// Measure MacipACLInterfaceAddDel time.
+		start := time.Now()
+
+		req := &acl_api.MacipACLInterfaceAddDel{
+			ACLIndex:  removedACLIndex,
+			SwIfIndex: ifIdx,
+			IsAdd:     0,
+		}
+
+		reply := &acl_api.MacipACLInterfaceAddDelReply{}
+
+		err := handler.callsChannel.SendRequest(req).ReceiveReply(reply)
+		if err != nil {
+			return fmt.Errorf("failed to remove L2 ACL %v from interface %v", removedACLIndex, ifIdx)
```

Comment:
You are ignoring the `err` value here, it should be included in the error message.

Also the word order in this error message is reversed to the case below it. It should start the same: `remove L2 ACL... ` and end with `.. failed: ` and then the original error message.